### ISSUE TITLE
Panic when running terraform plan with azurerm provider and a single resource

### ISSUE THREAD

**Issue by allanlang** [hashicorp/terraform#6270]
### Terraform Version
Terraform v0.6.14 / Mac OS 10.11.4

### Affected Resource(s)
There is a single configured resource besides the `azurerm` provider:
- `azurerm_resource_group "espoc"`

### Terraform Configuration Files
```hcl
provider "azurerm" {
  subscription_id = "{var.subscription_id}"
  client_id       = "{var.client_id}"
  client_secret   = "{var.client_secret}"
  tenant_id       = "{var.tenant_id}"
}

resource "azurerm_resource_group" "espoc" {
    name     = "espoc"
    location = "{var.region}"
}
```

### Panic Output
See https://gist.github.com/allanlang/0adc614c9f441cf96e3538be8e37c788 for `crash.log` output and associated config files (as above).

### Expected Behavior
Running `terraform plan` should describe the changes to be made, in this case the creation of a single resource group.

### Actual Behavior
Terraform crashes with error message including `panic: runtime error: invalid memory address or nil pointer dereference`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Prepare a variables.tf file providing valid values for subscription_id, client_it, client_secret, tenant_id and region. Place this file alongside another file containing the configuration provided above.
2. `terraform plan`

### Important Factoids
None

**Comment by stack72** [comment]
Hi @allanlang 

It looks to me like the credentials are incorrect. We have had a number of instances of this

```
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: panic: runtime error: invalid memory address or nil pointer dereference
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: [signal 0xb code=0x1 addr=0x10 pc=0x280857]
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: 
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: goroutine 38 [running]:
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: panic(0x925c00, 0xc82000a0b0)
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: 	/opt/go/src/runtime/panic.go:464 +0x3e6
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: github.com/hashicorp/terraform/vendor/github.com/Azure/azure-sdk-for-go/arm/resources/resources.ProvidersClient.Register(0x1249498, 0xc8201fe6c0, 0x12494c0, 0xc8202aa660, 0x0, 0x0, 0xaf6160, 0x13, 0x0, 0xd18c2e2800, ...)
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: 	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/Azure/azure-sdk-for-go/arm/resources/resources/providers.go:206 +0x217
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: github.com/hashicorp/terraform/builtin/providers/azurerm.registerAzureResourceProvidersWithSubscription(0xc8201f6190, 0xc8202ae000, 0x0, 0x0)
2016/04/20 22:59:05 [DEBUG] terraform-provider-azurerm: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/azurerm/provider.go:150 +0x140
```

There is a section in the [docs](https://www.terraform.io/docs/providers/azurerm/index.html) about creating the credentials

Can you check that the credentials are correct? We found that sometimes credentials already created were not working for other people

Paul

**Comment by stack72** [comment]
Hi @allanlang 

On looking at this again, your config isn't quite right:

```
provider "azurerm" {
  subscription_id = "{var.subscription_id}"
  client_id       = "{var.client_id}"
  client_secret   = "{var.client_secret}"
  tenant_id       = "{var.tenant_id}"
}
```

should be:

```
provider "azurerm" {
  subscription_id = "${var.subscription_id}"
  client_id       = "${var.client_id}"
  client_secret   = "${var.client_secret}"
  tenant_id       = "${var.tenant_id}"
}
```
We are not actually calling the interpolation - we are passing those strings directly to the provider. This has still showed that there is a bug - that should fail gracefully, not panic

We will work on that

Paul

**Comment by allanlang** [comment]
@stack72 you are quite right, the error was all mine. First I had a non-parameterised configuration where I had simply messed up the client secret. When I refactored this to make it suitable for publishing here, I missed the $ symbol. Net result both times - junk passed to the provider's configuration.

Improved error handling would be great, but apart from that apologies for wasting your time!

**Comment by stack72** [comment]
Hi @allanlang 

Glad its worked outm you haven't wasted any time at all - its good to point out the unnecessary panic in this situation. We are actively trying to fix that right now

Paul

**Comment by stack72** [comment]
Fixed in #6290
